pull: fix PullFlags numbering

lnussel opened the pull request:
Without this change, systemd-sysupdate --verify=no would still fail with
```
Literal checksum verification only supported if no associated files are downloaded.
```

poettering commented:
hmm, if we have a hash value we should verify it in all cases.

if you get the error `Literal checksum verification only supported if no associated files are downloaded.` then there's some logic problem elsewhere. I.e. we should not be executing the pull commands while requesting associated files. So the bug is that for some reason associated files are requested even though they should not be.

Can you paste the full debug log output of this run somehwere along we the precise used command line

lnussel commented:
```
# SYSTEMD_LOG_LEVEL=debug ./systemd-sysupdate --verify=no update
Discovering installed instances…
Discovering available instances…
Manifest web cache miss for http://xxx.suse.de/lnussel/sysupdate/.
⤵️ Acquiring manifest file http://xxx.suse.de/lnussel/sysupdate/SHA256SUMS…
Successfully forked off '(sd-pull)' as PID 28591.
Writing downloaded data to STDOUT.
Pulling 'http://xxx.suse.de/lnussel/sysupdate/SHA256SUMS'.
varlink: Setting state idle-client
varlink: Sending message: {"method":"io.systemd.Resolve.ResolveHostname","parameters":{"name":"xxx.suse.de","flags":0}}
varlink: Changing state idle-client → calling
varlink: New incoming message: {"parameters":{"addresses":[{"ifindex":2,"family":2,"address":[10,xx,xx,xx]}],"name":"xxx.suse.de","flags":1048577}}
varlink: Changing state calling → called
varlink: Changing state called → idle-client
*   Trying 10.xx.xx.xx:80...
* Connected to xxx.suse.de (10.xx.xx.xx) port 80 (#0)
> GET /lnussel/sysupdate/SHA256SUMS HTTP/1.1
Host: xxx.suse.de
User-Agent: systemd-pull/251+git20220718.98c143edc
Accept: */*

* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Date: Thu, 04 Aug 2022 06:53:58 GMT
< Server: Apache
< Last-Modified: Tue, 19 Jul 2022 11:10:35 GMT
< ETag: "110-5e42688a068a2"
< Accept-Ranges: bytes
< Content-Length: 272
Downloading 272B for http://xxx.suse.de/lnussel/sysupdate/SHA256SUMS.
< 
Stream is compressed: uncompressed
* Connection #0 to host xxx.suse.de left intact
Acquired 272B.
Download of http://xxx.suse.de/lnussel/sysupdate/SHA256SUMS complete.
Combined progress 80%
Combined progress 90%
Operation completed successfully.
Exiting.
(sd-pull) succeeded.
Added manifest 'http://xxx.suse.de/lnussel/sysupdate/' to cache.
Determining installed update sets…
Determining available update sets…
Selected update '20220714' for install.
Making room for 1 updates…
Removed no instances.
⤵️ Acquiring http://xxx.suse.de/lnussel/sysupdate/opensuse-tumbleweed-image.ix86-1.0.0-networkd-Snapshot20220714.tar.xz → /var/lib/machines/tw-networkd_20220714...
Successfully forked off '(sd-pull-tar)' as PID 28593.
Literal checksum verification only supported if no associated files are downloaded.
(sd-pull-tar) failed with exit status 1.
# cat /etc/sysupdate.d/tw-networkd.conf 
[Source]
Type=url-tar
Path=http://xxx.suse.de/lnussel/sysupdate/
MatchPattern=opensuse-tumbleweed-image.ix86-1.0.0-networkd-Snapshot@v.tar.xz 

[Target]
Type=subvolume
Path=/var/lib/machines
MatchPattern=tw-networkd_@v
CurrentSymlink=tw-networkd
```

lnussel commented:
Here's the real issue, those flags are messed up https://github.com/systemd/systemd/blob/main/src/import/pull-common.h#L9

poettering commented:
Ouch! That looks like a fuck-up I did while rebasing. Yuck!

medhefgo commented:
This definitely needs a stable backport.

poettering commented:
man, i wished gcc/clang would optionally complain if you define to enum values to the same literal value within the same enum{} construct. i.e. so that this would cause a wrning:

```c
enum {
    A = 7,
    B = 7,
};
```

but this would not:

```c
enum {
    A = 7,
    B = A,
};
```
